### PR TITLE
Feature/core 936 add proof to vc

### PR DESCRIPTION
**Pull request by wulfraem**
- [CORE-936]

[CORE-936]: https://evannetwork.atlassian.net/browse/CORE-936

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,13 +43,165 @@ Allows to work with VCs on [evan.network], currently includes:
 - validating VCs, which will
   - check `proof` (if attached)
   - check `credentialStatus` online (if attached)
+- creating VCs
+
+#### Retrieving VCs
+
+Existing VCs from [evan.network] can be retrieved with vade's [`get_vc_document] function:
```
Comment (Tschuck) [inline README.md:50]:
```suggestion
Existing VCs from [evan.network] can be retrieved with vade's [`get_vc_document`] function:
```

File: README.md [hunk 2]
```diff
@@ -43,13 +43,165 @@ Allows to work with VCs on [evan.network], currently includes:
...
+
+Taken from our tests:
+
+```rust
+use vade::Vade;
+use vade::plugin::rust_storage_cache::RustStorageCache;
+use vade_evan::plugin::rust_vcresolver_evan::RustVcResolverEvan;
+
+async fn example() -> std::result::Result<(), Box<dyn std::error::Error>> {
+    // create a vc resolver with attached did resolver in a vade
```
Comment (Tschuck) [inline README.md:81]:
```suggestion
    // create a vc resolver with attached did resolver
```

File: README.md [hunk 3]
```diff
@@ -43,13 +43,165 @@ Allows to work with VCs on [evan.network], currently includes:
...
+    let mut vcr_vade = Vade::new();
+    vcr_vade.register_did_resolver(Box::from(vcr_didr));
+    // add did document to vcr's did resolver
+    let did_id = "did:evan:testcore:0x0ef0e584c714564a4fc0c6c367edccb0c1cbf65f";
+    let did_document = "{...}";
+    vcr_vade.set_did_document(&did_id, &did_document).await?;
+    let mut vcr = RustVcResolverEvan::new();
+    vcr.vade = Some(Box::from(vcr_vade));
+
+    // create vade to work with, attach
```
Comment (Tschuck) [inline README.md:92]:
```suggestion
    // create vade to work with, attach vc resolver
```

File: README.md [hunk 4]
```diff
@@ -43,13 +43,165 @@ Allows to work with VCs on [evan.network], currently includes:
...
+- register it as `vade` in our [`RustVcResolver`] instance
+
+This allows us to validate the `proof` property in our VC document.
+
+
+##### Creating VCs
+
+Creating a VC currently has three requirements:
+
+- and [evan.network] identity for the VC issuer, which means, we also have
```
Comment (Tschuck) [inline README.md:120]:
```suggestion
- an [evan.network] identity for the VC issuer, which means, we also have
```

File: README.md [hunk 5]
```diff
@@ -43,13 +43,165 @@ Allows to work with VCs on [evan.network], currently includes:
...
+  - an DID document for for the issuer of our VC
```
Comment (Tschuck) [inline README.md:121]:
```suggestion
  - an DID document for the issuer of our VC
```

File: README.md [hunk 6]
```diff
@@ -43,13 +43,165 @@ Allows to work with VCs on [evan.network], currently includes:
...
+    let veri_pkey = "01734663843202e2245e5796cb120510506343c67915eb4f9348ac0d8c2cf22a";
+    // sample data, `id` is required, `credentialSubject` is optional and holds tests data
+    let partial_vc_data =r###"
+    {
+        "id": "foo-bar-vc",
+        "credentialSubject": {
+            "foo": "bar"
+        }
+    } 
+"###;
```
Comment (Tschuck) [inline README.md:151]: Is this format correct? :D

File: src/lib.rs [hunk 7]
```diff
@@ -51,18 +51,178 @@
 //! - validating VCs, which will
 //!   - check `proof` (if attached)
 //!   - check `credentialStatus` online (if attached)
+//! - creating VCs
+//!
+//! #### Retrieving VCs
+//!
+//! Existing VCs from [evan.network] can be retrieved with vade's [`get_vc_document] function:
```
Comment (Tschuck) [inline src/lib.rs:58]:
```suggestion
//! Existing VCs from [evan.network] can be retrieved with vade's [`get_vc_document`] function:
```

File: src/lib.rs [hunk 8]
```diff
@@ -51,18 +51,178 @@
...
+//!
+//! Taken from our tests:
+//!
+//! ```rust
+//! use vade::Vade;
+//! use vade::plugin::rust_storage_cache::RustStorageCache;
+//! use vade_evan::plugin::rust_vcresolver_evan::RustVcResolverEvan;
+//!
+//! async fn example() -> std::result::Result<(), Box<dyn std::error::Error>> {
+//!     // create a vc resolver with attached did resolver in a vade
```
Comment (Tschuck) [inline src/lib.rs:91]:
```suggestion
//!     // create a vc resolver with attached did resolver
```

File: src/lib.rs [hunk 9]
```diff
@@ -51,18 +51,178 @@
...
+//!     let mut vcr_vade = Vade::new();
+//!     vcr_vade.register_did_resolver(Box::from(vcr_didr));
+//!     // add did document to vcr's did resolver
+//!     let did_id = "did:evan:testcore:0x0ef0e584c714564a4fc0c6c367edccb0c1cbf65f";
+//!     let did_document = "{...}";
+//!     vcr_vade.set_did_document(&did_id, &did_document).await?;
+//!     let mut vcr = RustVcResolverEvan::new();
+//!     vcr.vade = Some(Box::from(vcr_vade));
+//!
+//!     // create vade to work with, attach
```
Comment (Tschuck) [inline src/lib.rs:102]:
```suggestion
//!     // create vade to work with, attach rc resolver
```

File: src/lib.rs [hunk 10]
```diff
@@ -51,18 +51,178 @@
...
+//! - register it as `vade` in our [`RustVcResolver`] instance
+//!
+//! This allows us to validate the `proof` property in our VC document.
+//!
+//!
+//! ##### Creating VCs
+//!
+//! Creating a VC currently has three requirements:
+//!
+//! - and [evan.network] identity for the VC issuer, which means, we also have
```
Comment (Tschuck) [inline src/lib.rs:130]:
```suggestion
//! - an [evan.network] identity for the VC issuer, which means, we also have
```

File: src/lib.rs [hunk 11]
```diff
@@ -51,18 +51,178 @@
...
+//!   - an DID document for for the issuer of our VC
```
Comment (Tschuck) [inline src/lib.rs:131]:
```suggestion
//!   - an DID document for the issuer of our VC
```

File: src/plugin/rust_vcresolver_evan.rs [hunk 12]
```diff
@@ -98,6 +106,69 @@ impl RustVcResolverEvan {
...
+        if !parsed_vc["@context"].as_array().unwrap().iter().any(|v| v == VC_W3C_MANDATORY_CONTEXT) {
+            parsed_vc["@context"].as_array_mut().unwrap().push(Value::from(VC_W3C_MANDATORY_CONTEXT));
+        }
+
+        // ensure type
+        if parsed_vc["type"].is_null() {
+            parsed_vc["type"] = Value::from(VC_DEFAULT_TYPE);
+        }
+
+        // if not privided, fill issuer with given `verification_methoda`, did
```
Comment (Tschuck) [inline src/plugin/rust_vcresolver_evan.rs:149]:
```suggestion
        // if not privided, fill issuer with given `verification_method`, did
```

File: README.md (hunk 2 above)
Comment (wulfraem) [reply README.md:81]: That was intended, updated it to `vade instance` to make it less confusing.

File: README.md (hunk 4 above)
Comment (wulfraem) [reply README.md:120]: fixed it

File: README.md (hunk 5 above)
Comment (wulfraem) [reply README.md:121]: fixed

File: README.md (hunk 6 above)
Comment (wulfraem) [reply README.md:151]: correct, but not pretty, updated format

File: src/lib.rs (hunk 7 above)
Comment (wulfraem) [reply src/lib.rs:58]: fixed

File: src/lib.rs (hunk 8 above)
Comment (wulfraem) [reply src/lib.rs:91]: fixed

File: src/lib.rs (hunk 9 above)
Comment (wulfraem) [reply src/lib.rs:102]: fixed

File: src/lib.rs (hunk 10 above)
Comment (wulfraem) [reply src/lib.rs:130]: fixed

File: src/lib.rs (hunk 11 above)
Comment (wulfraem) [reply src/lib.rs:131]: fixed

**Pull request merged**